modules/standard/IO.chpl

```diff
@@ -1971,7 +1961,19 @@ proc =(ref ret:channel, x:channel) {
 }
 
 pragma "no doc"
-proc channel.channel(param writing:bool, param kind:iokind, param locking:bool, f:file, out error:syserr, hints:c_int, start:int(64), end:int(64), in local_style:iostyle) {
+proc channel.init(param writing:bool, param kind:iokind, param locking:bool) {
+  this.writing = writing;
+  this.kind = kind;
+  this.locking = locking;
+  super.init();
+}
+
+pragma "no doc"
+proc channel.init(param writing:bool, param kind:iokind, param locking:bool, f:file, out error:syserr, hints:c_int, start:int(64), end:int(64), in local_style:iostyle) {
+  this.writing = writing;
+  this.kind = kind;
+  this.locking = locking;
+  super.init();
```

Comment:
I would be tempted to have one of these initializers call the other one instead of having them both utilize the same Phase 1 code.  (Likely having the bottom one call the top one, though I would check with @mppf to see if the top initializer is okay or if it should really be calling into the bottom one in some form).

Reply:
(Update for those following along at home: the above strategy currently encounters a bug for generic records, see #7960)